### ISSUE TITLE
Error while trying to launch appium for android: UiAutomator quit before it successfully launched

### ISSUE THREAD

**Issue by mzbedat** [appium/appium#6474]
## The problem

I'm getting the following error when trying to open android application (from a robotframework project with AppiumLibrary installed):
Original error: UiAutomator quit before it successfully launched

Tried with different devices with different android versions but still getting the same.
## Environment
- Appium version: 1.4.16.1
- Desktop OS/version used to run Appium: Windows 10
- Mobile platform/version under test: Android (6.0 / 5.0.1 / 4.4/ ...)
- Real device (Galaxy S5 / Galaxy S3 / LG / ..)
- Appium.app|exe:
## Details

In Appium log I can see :
1) can find uiautomator process (kill all command)
2) IllegalStateException while running bootstrap:

I found another -old- issue that seems relevant to me:
[https://github.com/appium/appium/issues/1517](url)

> info: Starting App
> info: [debug] Attempting to kill all 'uiautomator' processes
> info: [debug] Getting all processes with 'uiautomator'
> info: [debug] executing cmd: "C:\Program Files (x86)\Android\android-sdk\platform-tools\adb.exe" -s 5e1858cc shell "ps 'uiautomator'"
> info: [debug] No matching processes found
> 
> info: [debug] Running bootstrap
> info: [debug] spawning: C:\Program Files (x86)\Android\android-sdk\platform-tools\adb.exe -s 5e1858cc shell uiautomator runtest AppiumBootstrap.jar -c io.appium.android.bootstrap.Bootstrap -e pkg com.netflix.mediaclient -e disableAndroidWatchers false
> info: [debug] [UIAUTOMATOR STDOUT] INSTRUMENTATION_RESULT: shortMsg=java.lang.IllegalStateException
> info: [debug] [UIAUTOMATOR STDOUT] INSTRUMENTATION_RESULT: longMsg=UiAutomationService android.accessibilityservice.IAccessibilityServiceClient$Stub$Proxy@f24638already registered!
> info: [debug] [UIAUTOMATOR STDOUT] INSTRUMENTATION_CODE: 0
> info: [debug] UiAutomator exited
## Link to Appium logs

[https://gist.github.com/mzbedat/5dd7fb0b5f650022eb0e819e3215efeb](url)

<script src="https://gist.github.com/mzbedat/5dd7fb0b5f650022eb0e819e3215efeb.js"></script>

## Code To Reproduce Issue [ Good To Have ]

Robot code:

`**\* Settings ***

Documentation  Keywords Resources
Library     uiautomatorlibrary
Library     AppiumLibrary   #10 ##Sets default timeout to 10 seconds

**\* Test Cases **\*  

Play Netflix with Appium
    #Start Application  com.netflix.mediaclient/.ui.launch.UIWebViewActivity

```
####### Open Application | http://localhost:4723/wd/hub | platformName=Android | platformVersion=4.2.2 | deviceName=192.168.56.101:5555 | app=${CURDIR}/demoapp/OrangeDemoApp.apk | appPackage=com.netease.qa.orangedemo | appActivity=MainActivity |
# For wifi connection # deviceName=192.168.56.101:5555
# Use appPackage and appActivity if your apk is already on device. 
# Use app if you want Appium to install the apk on every launch.


Open Application    http://localhost:4723/wd/hub    platformName=Android    deviceName='Galaxy S5'  app=C:\\workspace_peerapp\\PeerApp_Automation\\resources\\APK\\Netflix-5.0.apk  ##appPackage=com.netflix.mediaclient    appActivity = appActivity=MainActivity   

#Open Application   http://localhost:4723/wd/hub    platformName=Android    platformVersion=6.0 deviceName='Galaxy S5'  app=C:\\workspace_peerapp\\PeerApp_Automation\\resources\\APK\\google.apk   ##appPackage=com.google.android.apps.translate  appActivity = TranslateActivity`
```

**Comment by imurchie** [comment]
Can you run the spawn code yourself?

```
C:\Program Files (x86)\Android\android-sdk\platform-tools\adb.exe -s 5e1858cc shell uiautomator runtest AppiumBootstrap.jar -c io.appium.android.bootstrap.Bootstrap -e pkg com.netflix.mediaclient -e disableAndroidWatchers false
```

The `AppiumBootstrap.jar` file ought to be in the `build/android_bootstrap` directory in your Appium installation.

**Comment by SrinivasanTarget** [comment]
@triager please label: Android

**Comment by mzbedat** [comment]
Hi,
I ran the following command:


C:\Program Files (x86)\Appium\node_modules\appium\build\android_bootstrap>adb.exe -s 5e1858cc shell uiautomator runtest AppiumBootstrap.jar -c io.appium.android.bootstrap.Bootstrap -e pkg com.netflix.mediaclient -e disableAndroidWatchers false



INSTRUMENTATION_STATUS: numtests=1

INSTRUMENTATION_STATUS: stream=

io.appium.android.bootstrap.Bootstrap:

INSTRUMENTATION_STATUS: id=UiAutomatorTestRunner

INSTRUMENTATION_STATUS: test=testRunServer

INSTRUMENTATION_STATUS: class=io.appium.android.bootstrap.Bootstrap

INSTRUMENTATION_STATUS: current=1

INSTRUMENTATION_STATUS_CODE: 1

[APPIUM-UIAUTO] [debug] Socket opened on port 4724[/APPIUM-UIAUTO]

[APPIUM-UIAUTO] [debug] Appium Socket Server Ready[/APPIUM-UIAUTO]

[APPIUM-UIAUTO] [debug] Loading json...[/APPIUM-UIAUTO]

[APPIUM-UIAUTO] [debug] json loading complete.[/APPIUM-UIAUTO]

[APPIUM-UIAUTO] [debug] Registered crash watchers.[/APPIUM-UIAUTO]



C:\Program Files (x86)\Appium\node_modules\appium\build\android_bootstrap>


(Then ran the test again (with appium server up) and got the same error.)

B.R

**Comment by mzbedat** [comment]
Hi,
Any advise please...

**Comment by eliasnogueira** [comment]
I have the same problem, but I know when it occurs, but I don't know how to solve it....

**My error scenario**
- I have a grid with 12 devices execution tests in parallel
- When any error (like NoSuchElement ou TimeoutException) occurs there's an implementation to try execute the test again
- In this second execution (test again) I got the error UiAutomator quit before it successfully launched

**What I've tried**
If I shut down the grid and nodes and run again, the problem not occurs (but comes back when any exception throws lather)


**Question**
Is there a way to force close uiautomator before execute a test on a node?
I ran the command to verify uiautmoator [1] and it returns uitomator error


**[1] Command and error**
_Command_
`adb -s 0014659290 shell uiautomator runtest AppiumBootstrap.jar -c io.appium.android.bootstrap.Bootstrap -e pkg com.netflix.mediaclient -e disableAndroidWatchers false`

_Error_
```
C:\Users\elias>adb -s 0014659290 shell uiautomator runtest AppiumBootstrap.jar -c io.appium.android.bootstrap.Bootstrap -e pkg com.netflix.mediaclient -e dis ableAndroidWatchers false
INSTRUMENTATION_RESULT: shortMsg=java.lang.IllegalStateException
INSTRUMENTATION_RESULT: longMsg=UiAutomationService android.accessibilityservice
.IAccessibilityServiceClient$Stub$Proxy@8363ab8already registered!
INSTRUMENTATION_CODE: 0
```

Any suggestions @jlipps @SrinivasanTarget @bootstraponline ?
Regards!

**Comment by harlentan** [comment]
I got the same issue. I also got some advise to kill uiautomator use adb shell.

But this seems do not work well, I get the same error   after the uiautomator killed on device.

> .IAccessibilityServiceClient$Stub$Proxy@8363ab8already registered!

maybe there is some other service started by uiautomator that should be terminated before running command:

> adb -s 0014659290 shell uiautomator runtest AppiumBootstrap.jar -c io.appium.android.bootstrap.Bootstrap -e pkg com.netflix.mediaclient -e disableAndroidWatchers false

I have been stuck in this issue for a couple of days.

does anyone know how to solve this issue?
